CriticalMass/NetworkOperator.swift

```diff
@@ -48,4 +48,8 @@ struct NetworkOperator: NetworkLayer {
         }
         task.resume()
     }
+
+    func cancelActiveRequestIfNeeded() {
+        session.invalidateAndCancel()
```

Comment:
Works but maybe the `NetworkOperator` could return the current task and you just cancel that in the `RequestManager`. Though the Managers do not share a session, right?

Reply:
We currently only have one Manager instance and it uses the same session for all requests. Changed the method name to requests to clarify that this method cancels all active requests

Reply:
Though wouldn't canceling the session while in the chatController also cancel tasks that are being performed from the MapViewController?

Reply:
Yes, so this method should only get called if you really want to cancel all tasks